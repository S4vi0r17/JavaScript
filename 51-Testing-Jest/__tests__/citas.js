import Citas from '../js/classes/Citas';

describe('Probar la clase de Citas', () => {

    const citas = new Citas();

    const id = Date.now();

    test('Agregar una nueva cita', () => {
        const citaObj = {
            id,
            mascota: 'Minino',
            propietario: 'Eder',
            telefono: '123456789',
            fecha: '10-12-2024',
            hora: '10:30',
            sintomas: 'Solo duerme'
        };

        citas.agregarCita(citaObj);

        // Prueba
        expect(citas).toMatchSnapshot();
    });

    test('Actualizar cita', () => {
        const citaActualizada = {
            id,
            mascota: 'New Minino',
            propietario: 'New Eder',
            telefono: '123456789',
            fecha: '10-12-2020',
            hora: '10:30',
            sintomas: 'Solo duerme'
        };

        citas.editarCita(citaActualizada);

        expect(citas).toMatchSnapshot();
    });

    test('Eliminar cita', () => {
        citas.eliminarCita(id);

        expect(citas).toMatchSnapshot();
    });

    test('Mostrar un mensaje de error cuando no hay citas', () => {
        const mensaje = citas.citas;

        expect(mensaje).toMatchSnapshot();
    });

});

// Para actualizar el snapshot, ejecutar el comando:
// npm test -- -u