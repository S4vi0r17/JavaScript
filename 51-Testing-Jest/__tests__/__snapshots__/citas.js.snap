// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probar la clase de Citas Actualizar cita 1`] = `
Citas {
  "citas": Array [
    Object {
      "fecha": "10-12-2020",
      "hora": "10:30",
      "id": 1705098870386,
      "mascota": "New Minino",
      "propietario": "New Eder",
      "sintomas": "Solo duerme",
      "telefono": "123456789",
    },
  ],
}
`;

exports[`Probar la clase de Citas Agregar una nueva cita 1`] = `
Citas {
  "citas": Array [
    Object {
      "fecha": "10-12-2024",
      "hora": "10:30",
      "id": 1705098870386,
      "mascota": "Minino",
      "propietario": "Eder",
      "sintomas": "Solo duerme",
      "telefono": "123456789",
    },
  ],
}
`;

exports[`Probar la clase de Citas Eliminar cita 1`] = `
Citas {
  "citas": Array [],
}
`;

exports[`Probar la clase de Citas Mostrar un mensaje de error cuando no hay citas 1`] = `Array []`;
